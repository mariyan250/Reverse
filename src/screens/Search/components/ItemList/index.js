// React and style
import React from 'react';
import './index.scss';

// Translations
import { useTranslation } from 'react-i18next';

// Components
import Item from '../Item';

function ItemList({ items }) {
  const { t } = useTranslation();

  return (
    <section className="item-list px-4">
      <h2>{t('search_screen.heading')}</h2>
      <ul className="app-bg-gray-darker pt-5 px-4">
        {items && items.map((item) => <Item {...item} key={item.id} />)}
      </ul>
    </section>
  );
}

export default ItemList;
