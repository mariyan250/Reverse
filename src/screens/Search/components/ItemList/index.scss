@import '../../../../index.scss';

.item-list {
  & > h2 {
    margin-top: 6.5rem !important;
    margin-bottom: 2rem !important;
  }

  ul {
    background: $background-gray-darker;
  }
}
