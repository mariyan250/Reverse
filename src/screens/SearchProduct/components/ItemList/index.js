// React and style
import React from 'react';
import './index.scss';

// Translations
import { useTranslation } from 'react-i18next';

// Components
import Item from '../Item';

function ItemList({ items, title }) {
  const { t } = useTranslation();

  return (
    <section className="item-list px-4">
      <h1 className="text-center">{t(title)}</h1>
      <ul className="app-bg-gray-darker pt-5 px-4">
        {items && items.map((item) => <Item {...item} key={item.id} />)}
      </ul>
    </section>
  );
}

export default ItemList;
