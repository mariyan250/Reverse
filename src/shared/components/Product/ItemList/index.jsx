// React and style
import React from 'react';
import './index.scss';

// Components
import Item from '../Item';

function ItemList({ items }) {
  return (
    <ul className="item-list">{items && items.map((item) => <Item {...item} key={item.id} />)}</ul>
  );
}

export default ItemList;
