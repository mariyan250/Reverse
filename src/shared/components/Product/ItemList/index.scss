@import '../../../../index.scss';

.item-list {
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  gap: to-rem(24px);

  @include media-breakpoint-sm-up {
    grid-template-columns: repeat(2, 1fr);
    row-gap: to-rem(24px);
  }

  @include media-breakpoint-md-up {
    grid-template-columns: repeat(3, 1fr);
  }

  @include media-breakpoint-lg-up {
    grid-template-columns: repeat(4, 1fr);
  }

  @include media-breakpoint-xl-up {
    grid-template-columns: repeat(5, 1fr);
  }
}
