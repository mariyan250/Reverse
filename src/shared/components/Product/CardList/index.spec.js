import React from 'react';
import { renderWithRouter } from 'tests/test-utils';
import ItemList from './index';

describe('Search', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  describe('Rendering', () => {
    it('should render the component with initial state', () => {
      const { asFragment } = renderWithRouter(<ItemList />, initialState);

      expect(asFragment(<ItemList />)).toMatchSnapshot();
    });
  });
});
