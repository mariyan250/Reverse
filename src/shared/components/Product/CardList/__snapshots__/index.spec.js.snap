// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Rendering should render the component with initial state 1`] = `
<DocumentFragment>
  <ul
    class="item-list"
  />
</DocumentFragment>
`;
